_pages/fsa/dynamicforms/core/basic-form-validations.md

```diff
@@ -0,0 +1,285 @@
+JSON example:
+```typescript
+"validations": [
+    {
+    "name": "containsValue",
+    "arguments": [
+        {
+        "value": ["test string"]
+        }
+    ]
+    }
+]       
+```
+- **compareNumbers** - Compares input value to other field. Accepts two arguments:
+  - name of the field to compare (string)
+  - operator (string) can be 'shouldBeGreater' or 'shouldBeLess' 
+   
+JSON example:
+```typescript
+"validations": [
+    {
+        "name": "compareNumbers",
+        "arguments": [
+        {
+            "value": "propertyValue"
+        },
+        {
+            "value": "shouldBeLess"
+        }
+        ]
+    }
+],
+```
+- **email** - Checks if input data is valid email
+- **alphanumeric** - Allows only numbers and letters as input data 
+
+## How all this works
+
+Before Form gets rendered, all validations from JSON are processed and in case they have corresponding implementation defined  in config they are transferred to mapped **ValidationFn**. Name from validation object corresponds the name defined in form configuration on SPA application. In this particular case it will be validation with name compareToCurrentDate :
+
+```typescript
+DynamicFormsConfig
+validators: {
+  ....
+  compareToCurrentDate: {
+    validator:  DefaultFormValidators.compareToCurrentDate
+  },
+  ....
+}
+```
+
+All arguments defined in JSON file represents inputs parameters for compareToCurrentDate function from DefaultFormValidators and they are added to the function in run-time. In our example based on mapping defined in configuration we will find following validation function:
+
+```typescript
+static compareToCurrentDate(operator) {
+   return (control: AbstractControl): ValidationErrors | null => {
+     const inputVal = new Date(control.value as string);
+     const today = new Date();
+     switch (operator) {
+       case 'shouldBeGreater':
+         return inputVal.getTime() > today.getTime()
+           ? null
+           : { InvalidDate: true };
+       case 'shouldBeLess':
+         return inputVal.getTime() < today.getTime()
+           ? null
+           : { InvalidDate: true };
+     }
+   };
+ }
+```
+and assign that function with input parameter 'shouldBeGreated' (operator in this case) to the corresponding field. In same way we can pass multiple input parameters via arguments attribute in case function requires more and add them to validation function respectively.
+
+## How to override existing validation in custom App
+
+In order to override some predefined validation, customers should be able to do that in their modules by referencing their custom functions with the same name as existing ones. Let's try to override compareToCurrentDate function, so that it just console logs customized message and returns always true for validation.  We will start in one of our modules by specifying following configuration:
```

Comment:
so that it just logs a customized message in the console and returns always true for validation.